analyzer.go

```diff
@@ -543,7 +544,8 @@ func (gosec *Analyzer) ParseErrors(pkg *packages.Package) error {
 // AppendError appends an error to the file errors
 func (gosec *Analyzer) AppendError(file string, err error) {
 	// Do not report the error for empty packages (e.g. files excluded from build with a tag)
-	if strings.Contains(err.Error(), "no buildable Go source files in") {
+	var noGoErr *build.NoGoError
+	if errors.As(err, &noGoErr) {
```

Comment:
Why not just checking with `errors.Is`:

```
if errors.Is(err, build.NoGoError) {
}
```

You don't need to create a temporary variable here since it not required to set its value, and later to use it in the context.

Reply:
It was my first thought to use `errors.Is`. But this can't compile:

<img width="674" alt="image" src="https://github.com/user-attachments/assets/f097cc13-e0f6-4179-9f25-8b301c3e7ff6" />

It's because [`build.NoGoError`](https://cs.opensource.google/go/go/+/refs/tags/go1.22.9:src/go/build/build.go;l=507) is defined as a struct:

```go
// NoGoError is the error used by [Import] to describe a directory
// containing no buildable Go source files. (It may still contain
// test files, files hidden by build tags, and so on.)
type NoGoError struct {
    Dir string
}

func (e *NoGoError) Error() string {
    return "no buildable Go source files in " + e.Dir
}

Reply:
Ok, that requires a sentinel error.